// ng2-toastr/src/toast-container.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ng2-toastr/src/toast-container.component';
import * as import3 from '@angular/platform-browser';
import * as import4 from 'ng2-toastr/src/toast-options';
var styles_ToastContainer = [];
export var RenderType_ToastContainer = import0.ɵcrt({
    encapsulation: 2,
    styles: styles_ToastContainer,
    data: { 'animation': [{
                name: 'inOut',
                definitions: [
                    {
                        type: 0,
                        name: 'flyRight, flyLeft',
                        styles: {
                            type: 6,
                            styles: {
                                opacity: 1,
                                transform: 'translateX(0)'
                            }
                        }
                    },
                    {
                        type: 0,
                        name: 'fade',
                        styles: {
                            type: 6,
                            styles: { opacity: 1 }
                        }
                    },
                    {
                        type: 0,
                        name: 'slideDown, slideUp',
                        styles: {
                            type: 6,
                            styles: {
                                opacity: 1,
                                transform: 'translateY(0)'
                            }
                        }
                    },
                    {
                        type: 1,
                        expr: 'void => flyRight',
                        animation: [
                            {
                                type: 6,
                                styles: {
                                    opacity: 0,
                                    transform: 'translateX(100%)'
                                }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '0.2s ease-in'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'flyRight => void',
                        animation: [{
                                type: 4,
                                styles: {
                                    type: 6,
                                    styles: {
                                        opacity: 0,
                                        transform: 'translateX(100%)'
                                    }
                                },
                                timings: '0.2s 10ms ease-out'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'void => flyLeft',
                        animation: [
                            {
                                type: 6,
                                styles: {
                                    opacity: 0,
                                    transform: 'translateX(-100%)'
                                }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '0.2s ease-in'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'flyLeft => void',
                        animation: [{
                                type: 4,
                                styles: {
                                    type: 6,
                                    styles: {
                                        opacity: 0,
                                        transform: 'translateX(-100%)'
                                    }
                                },
                                timings: '0.2s 10ms ease-out'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'void => fade',
                        animation: [
                            {
                                type: 6,
                                styles: { opacity: 0 }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '0.3s ease-in'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'fade => void',
                        animation: [{
                                type: 4,
                                styles: {
                                    type: 6,
                                    styles: { opacity: 0 }
                                },
                                timings: '0.3s 10ms ease-out'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'void => slideDown',
                        animation: [
                            {
                                type: 6,
                                styles: {
                                    opacity: 0,
                                    transform: 'translateY(-200%)'
                                }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '0.3s ease-in'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'slideDown => void',
                        animation: [{
                                type: 4,
                                styles: {
                                    type: 6,
                                    styles: {
                                        opacity: 0,
                                        transform: 'translateY(-200%)'
                                    }
                                },
                                timings: '0.3s 10ms ease-out'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'void => slideUp',
                        animation: [
                            {
                                type: 6,
                                styles: {
                                    opacity: 0,
                                    transform: 'translateY(200%)'
                                }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '0.3s ease-in'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'slideUp => void',
                        animation: [{
                                type: 4,
                                styles: {
                                    type: 6,
                                    styles: {
                                        opacity: 0,
                                        transform: 'translateY(200%)'
                                    }
                                },
                                timings: '0.3s 10ms ease-out'
                            }
                        ]
                    }
                ]
            }
        ] }
});
function View_ToastContainer_2(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'div', [[
                'class',
                'toast-close-button'
            ]
        ], null, [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('click' === en)) {
                var pd_0 = (co.removeToast(v.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        (l()(), import0.ɵted(null, ['×\n        ']))
    ], null, null);
}
function View_ToastContainer_3(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'div', [], [[
                8,
                'className',
                0
            ]
        ], null, null, null, null)),
        (l()(), import0.ɵted(null, [
            '',
            ''
        ]))
    ], null, function (ck, v) {
        var co = v.component;
        var currVal_0 = import0.ɵinlineInterpolate(1, '', (v.parent.context.$implicit.config.titleClass || co.titleClass), '');
        ck(v, 0, 0, currVal_0);
        var currVal_1 = v.parent.context.$implicit.title;
        ck(v, 1, 0, currVal_1);
    });
}
function View_ToastContainer_4(l) {
    return import0.ɵvid(0, [(l()(), import0.ɵeld(0, null, null, 0, 'span', [], [[
                8,
                'innerHTML',
                1
            ]
        ], null, null, null, null))], null, function (ck, v) {
        var co = v.component;
        var currVal_0 = co.sanitizer.bypassSecurityTrustHtml(v.parent.context.$implicit.message);
        ck(v, 0, 0, currVal_0);
    });
}
function View_ToastContainer_5(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [], [[
                8,
                'className',
                0
            ]
        ], null, null, null, null)),
        (l()(), import0.ɵted(null, [
            '',
            ''
        ]))
    ], null, function (ck, v) {
        var co = v.component;
        var currVal_0 = import0.ɵinlineInterpolate(1, '', (v.parent.context.$implicit.config.messageClass || co.messageClass), '');
        ck(v, 0, 0, currVal_0);
        var currVal_1 = v.parent.context.$implicit.message;
        ck(v, 1, 0, currVal_1);
    });
}
function View_ToastContainer_1(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 17, 'div', [], [
            [
                24,
                '@inOut',
                0
            ],
            [
                8,
                'className',
                0
            ]
        ], [
            [
                null,
                '@inOut.done'
            ],
            [
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('@inOut.done' === en)) {
                var pd_0 = (co.onAnimationEnd($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (co.clicked(v.context.$implicit) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)),
        (l()(), import0.ɵted(null, ['\n        '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_ToastContainer_2)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, [' \n        '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_ToastContainer_3)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n        '])),
        (l()(), import0.ɵeld(0, null, null, 8, 'div', [], null, null, null, null, null)),
        import0.ɵdid(8192, null, 0, import1.NgSwitch, [], { ngSwitch: [
                0,
                'ngSwitch'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n          '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_ToastContainer_4)),
        import0.ɵdid(139264, null, 0, import1.NgSwitchCase, [
            import0.ViewContainerRef,
            import0.TemplateRef,
            import1.NgSwitch
        ], { ngSwitchCase: [
                0,
                'ngSwitchCase'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n          '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_ToastContainer_5)),
        import0.ɵdid(8192, null, 0, import1.NgSwitchDefault, [
            import0.ViewContainerRef,
            import0.TemplateRef,
            import1.NgSwitch
        ], null, null),
        (l()(), import0.ɵted(null, ['\n        '])),
        (l()(), import0.ɵted(null, ['             \n      ']))
    ], function (ck, v) {
        var currVal_2 = v.context.$implicit.config.showCloseButton;
        ck(v, 3, 0, currVal_2);
        var currVal_3 = v.context.$implicit.title;
        ck(v, 6, 0, currVal_3);
        var currVal_4 = v.context.$implicit.config.enableHTML;
        ck(v, 9, 0, currVal_4);
        var currVal_5 = true;
        ck(v, 12, 0, currVal_5);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = co.animate;
        var currVal_1 = import0.ɵinlineInterpolate(1, 'toast toast-', v.context.$implicit.type, '');
        ck(v, 0, 0, currVal_0, currVal_1);
    });
}
export function View_ToastContainer_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵted(null, ['\n    '])),
        (l()(), import0.ɵeld(0, [[
                'toastContainer',
                1
            ]
        ], null, 4, 'div', [[
                'id',
                'toast-container'
            ]
        ], [
            [
                4,
                'position',
                null
            ],
            [
                8,
                'className',
                0
            ]
        ], null, null, null, null)),
        (l()(), import0.ɵted(null, ['\n      '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_ToastContainer_1)),
        import0.ɵdid(401408, null, 0, import1.NgForOf, [
            import0.ViewContainerRef,
            import0.TemplateRef,
            import0.IterableDiffers
        ], { ngForOf: [
                0,
                'ngForOf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n    '])),
        (l()(), import0.ɵted(null, ['\n    ']))
    ], function (ck, v) {
        var co = v.component;
        var currVal_2 = co.toasts;
        ck(v, 4, 0, currVal_2);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = co.position;
        var currVal_1 = import0.ɵinlineInterpolate(1, '', co.positionClass, '');
        ck(v, 1, 0, currVal_0, currVal_1);
    });
}
function View_ToastContainer_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'toast-container', [], null, null, null, View_ToastContainer_0, RenderType_ToastContainer)),
        import0.ɵdid(90112, null, 0, import2.ToastContainer, [
            import3.DomSanitizer,
            import0.ChangeDetectorRef,
            import0.NgZone,
            import4.ToastOptions
        ], null, null)
    ], null, null);
}
export var ToastContainerNgFactory = import0.ɵccf('toast-container', import2.ToastContainer, View_ToastContainer_Host_0, {}, {}, []);
