// ng2-toastr/src/toast-container.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ng2-toastr/src/toast-container.component';
import * as import3 from '@angular/platform-browser';
import * as import4 from 'ng2-toastr/src/toast-options';
const styles_ToastContainer:any[] = ([] as any[]);
export const RenderType_ToastContainer:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ToastContainer,
      data: {'animation': [{
        name: 'inOut',
        definitions: [
          {
            type: 0,
            name: 'flyRight, flyLeft',
            styles: {
              type: 6,
              styles: {
                opacity: 1,
                transform: 'translateX(0)'
              }

            }

          }
          ,
          {
            type: 0,
            name: 'fade',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 0,
            name: 'slideDown, slideUp',
            styles: {
              type: 6,
              styles: {
                opacity: 1,
                transform: 'translateY(0)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: 'void => flyRight',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'translateX(100%)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '0.2s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'flyRight => void',
              animation: [{
                type: 4,
                styles: {
                  type: 6,
                  styles: {
                    opacity: 0,
                    transform: 'translateX(100%)'
                  }

                }
                ,
                timings: '0.2s 10ms ease-out'
              }
            ]
          }
          ,
          {
            type: 1,
            expr: 'void => flyLeft',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'translateX(-100%)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '0.2s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'flyLeft => void',
              animation: [{
                type: 4,
                styles: {
                  type: 6,
                  styles: {
                    opacity: 0,
                    transform: 'translateX(-100%)'
                  }

                }
                ,
                timings: '0.2s 10ms ease-out'
              }
            ]
          }
          ,
          {
            type: 1,
            expr: 'void => fade',
            animation: [
              {
                type: 6,
                styles: {opacity: 0}
              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '0.3s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'fade => void',
              animation: [{
                type: 4,
                styles: {
                  type: 6,
                  styles: {opacity: 0}
                }
                ,
                timings: '0.3s 10ms ease-out'
              }
            ]
          }
          ,
          {
            type: 1,
            expr: 'void => slideDown',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'translateY(-200%)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '0.3s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'slideDown => void',
              animation: [{
                type: 4,
                styles: {
                  type: 6,
                  styles: {
                    opacity: 0,
                    transform: 'translateY(-200%)'
                  }

                }
                ,
                timings: '0.3s 10ms ease-out'
              }
            ]
          }
          ,
          {
            type: 1,
            expr: 'void => slideUp',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'translateY(200%)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '0.3s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'slideUp => void',
              animation: [{
                type: 4,
                styles: {
                  type: 6,
                  styles: {
                    opacity: 0,
                    transform: 'translateY(200%)'
                  }

                }
                ,
                timings: '0.3s 10ms ease-out'
              }
            ]
          }

        ]

      }
  ]}
}
);
function View_ToastContainer_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'toast-close-button'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.removeToast((<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['×\n        ']))
  ]
  ,(null as any),(null as any));
}
function View_ToastContainer_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',((<any>v.parent).context.$implicit.config.titleClass || co.titleClass),'');
    ck(v,0,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.title;
    ck(v,1,0,currVal_1);
  });
}
function View_ToastContainer_4(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[
        8,
        'innerHTML',
        1
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.sanitizer.bypassSecurityTrustHtml((<any>v.parent).context.$implicit.message);
    ck(v,0,0,currVal_0);
  });
}
function View_ToastContainer_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',((<any>v.parent).context.$implicit.config.messageClass || co.messageClass),'');
    ck(v,0,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.message;
    ck(v,1,0,currVal_1);
  });
}
function View_ToastContainer_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',([] as any[]),[
      [
        24,
        '@inOut',
        0
      ]
      ,
      [
        8,
        'className',
        0
      ]

    ]
    ,[
      [
        (null as any),
        '@inOut.done'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('@inOut.done' === en)) {
        const pd_0:any = ((<any>co.onAnimationEnd($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.clicked(v.context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ToastContainer_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[' \n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ToastContainer_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,(null as any),0,import1.NgSwitch,([] as any[]),{ngSwitch: [
        0,
        'ngSwitch'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ToastContainer_4)),
    import0.ɵdid(139264,(null as any),0,import1.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ToastContainer_5)),
    import0.ɵdid(8192,(null as any),0,import1.NgSwitchDefault,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['             \n      ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = v.context.$implicit.config.showCloseButton;
    ck(v,3,0,currVal_2);
    const currVal_3:any = v.context.$implicit.title;
    ck(v,6,0,currVal_3);
    const currVal_4:any = v.context.$implicit.config.enableHTML;
    ck(v,9,0,currVal_4);
    const currVal_5:any = true;
    ck(v,12,0,currVal_5);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.animate;
    const currVal_1:any = import0.ɵinlineInterpolate(1,'toast toast-',v.context.$implicit.type,'');
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export function View_ToastContainer_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,[[
        'toastContainer',
        1
      ]
      ],(null as any),4,'div',[[
        'id',
        'toast-container'
      ]
    ],[
      [
        4,
        'position',
        (null as any)
      ]
      ,
      [
        8,
        'className',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ToastContainer_1)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import2.ToastContainer = v.component;
    const currVal_2:any = co.toasts;
    ck(v,4,0,currVal_2);
  },(ck,v) => {
    var co:import2.ToastContainer = v.component;
    const currVal_0:any = co.position;
    const currVal_1:any = import0.ɵinlineInterpolate(1,'',co.positionClass,'');
    ck(v,1,0,currVal_0,currVal_1);
  });
}
function View_ToastContainer_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'toast-container',([] as any[]),(null as any),(null as any),(null as any),View_ToastContainer_0,RenderType_ToastContainer)),
    import0.ɵdid(90112,(null as any),0,import2.ToastContainer,[
      import3.DomSanitizer,
      import0.ChangeDetectorRef,
      import0.NgZone,
      import4.ToastOptions
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ToastContainerNgFactory:import0.ComponentFactory<import2.ToastContainer> = import0.ɵccf('toast-container',import2.ToastContainer,View_ToastContainer_Host_0,{},{},([] as any[]));
