// ng2-pagination/dist/pagination-controls.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ng2-pagination/dist/pagination-controls.component';
import * as import3 from 'ng2-pagination/dist/pagination-controls.directive';
import * as import4 from 'ng2-pagination/dist/pagination.service';
const styles_PaginationControlsComponent:any[] = ['.ng2-pagination[_ngcontent-%COMP%] {\n  margin-left: 0;\n  margin-bottom: 1rem; }\n  .ng2-pagination[_ngcontent-%COMP%]::before, .ng2-pagination[_ngcontent-%COMP%]::after {\n    content: \' \';\n    display: table; }\n  .ng2-pagination[_ngcontent-%COMP%]::after {\n    clear: both; }\n  .ng2-pagination[_ngcontent-%COMP%]   li[_ngcontent-%COMP%] {\n    -moz-user-select: none;\n    -webkit-user-select: none;\n    -ms-user-select: none;\n    font-size: 0.875rem;\n    margin-right: 0.0625rem;\n    border-radius: 0; }\n  .ng2-pagination[_ngcontent-%COMP%]   li[_ngcontent-%COMP%] {\n    display: inline-block; }\n  .ng2-pagination[_ngcontent-%COMP%]   a[_ngcontent-%COMP%], .ng2-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%] {\n    color: #0a0a0a; \n    display: block;\n    padding: 0.1875rem 0.625rem;\n    border-radius: 0; }\n    .ng2-pagination[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover, .ng2-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover {\n      background: #e6e6e6; }\n  .ng2-pagination[_ngcontent-%COMP%]   .current[_ngcontent-%COMP%] {\n    padding: 0.1875rem 0.625rem;\n    background: #2199e8;\n    color: #fefefe;\n    cursor: default; }\n  .ng2-pagination[_ngcontent-%COMP%]   .disabled[_ngcontent-%COMP%] {\n    padding: 0.1875rem 0.625rem;\n    color: #cacaca;\n    cursor: default; } \n    .ng2-pagination[_ngcontent-%COMP%]   .disabled[_ngcontent-%COMP%]:hover {\n      background: transparent; }\n  .ng2-pagination[_ngcontent-%COMP%]   .ellipsis[_ngcontent-%COMP%]::after {\n    content: \'…\';\n    padding: 0.1875rem 0.625rem;\n    color: #0a0a0a; }\n\n.ng2-pagination[_ngcontent-%COMP%]   .pagination-previous[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]::before, .ng2-pagination[_ngcontent-%COMP%]   .pagination-previous.disabled[_ngcontent-%COMP%]::before { \n  content: \'«\';\n  display: inline-block;\n  margin-right: 0.5rem; }\n\n.ng2-pagination[_ngcontent-%COMP%]   .pagination-next[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]::after, .ng2-pagination[_ngcontent-%COMP%]   .pagination-next.disabled[_ngcontent-%COMP%]::after {\n  content: \'»\';\n  display: inline-block;\n  margin-left: 0.5rem; }\n\n.ng2-pagination[_ngcontent-%COMP%]   .show-for-sr[_ngcontent-%COMP%] {\n  position: absolute !important;\n  width: 1px;\n  height: 1px;\n  overflow: hidden;\n  clip: rect(0, 0, 0, 0); }'];
export const RenderType_PaginationControlsComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_PaginationControlsComponent,
  data: {}
}
);
function View_PaginationControlsComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'a',[[
        'aria-label',
        'Next page'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov((<any>(<any>(<any>v.parent).parent).parent),2).previous()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                Previous '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'show-for-sr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['page'])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_PaginationControlsComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Previous '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'show-for-sr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['page']))
  ]
  ,(null as any),(null as any));
}
function View_PaginationControlsComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[[
        'class',
        'pagination-previous'
      ]
      ],[[
        2,
        'disabled',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[' \n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = (1 < import0.ɵnov((<any>(<any>v.parent).parent),2).getCurrent());
    ck(v,3,0,currVal_1);
    const currVal_2:any = import0.ɵnov((<any>(<any>v.parent).parent),2).isFirstPage();
    ck(v,6,0,currVal_2);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov((<any>(<any>v.parent).parent),2).isFirstPage();
    ck(v,0,0,currVal_0);
  });
}
function View_PaginationControlsComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'a',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov((<any>(<any>(<any>v.parent).parent).parent),2).setCurrent((<any>v.parent).context.$implicit.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'show-for-sr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Page'])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.label;
    ck(v,6,0,currVal_0);
  });
}
function View_PaginationControlsComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'show-for-sr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['You\'re on page'])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),[' \n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.label;
    ck(v,6,0,currVal_0);
  });
}
function View_PaginationControlsComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',([] as any[]),[[
        2,
        'current',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_6)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_7)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = (import0.ɵnov((<any>(<any>v.parent).parent),2).getCurrent() !== v.context.$implicit.value);
    ck(v,3,0,currVal_1);
    const currVal_2:any = (import0.ɵnov((<any>(<any>v.parent).parent),2).getCurrent() === v.context.$implicit.value);
    ck(v,6,0,currVal_2);
  },(ck,v) => {
    const currVal_0:any = (import0.ɵnov((<any>(<any>v.parent).parent),2).getCurrent() === v.context.$implicit.value);
    ck(v,0,0,currVal_0);
  });
}
function View_PaginationControlsComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'a',[[
        'aria-label',
        'Next page'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov((<any>(<any>(<any>v.parent).parent).parent),2).next()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                Next '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'show-for-sr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['page'])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_PaginationControlsComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Next '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'show-for-sr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['page']))
  ]
  ,(null as any),(null as any));
}
function View_PaginationControlsComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[[
        'class',
        'pagination-next'
      ]
      ],[[
        2,
        'disabled',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_9)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_10)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_1:boolean = !import0.ɵnov((<any>(<any>v.parent).parent),2).isLastPage();
    ck(v,3,0,currVal_1);
    const currVal_2:any = import0.ɵnov((<any>(<any>v.parent).parent),2).isLastPage();
    ck(v,6,0,currVal_2);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov((<any>(<any>v.parent).parent),2).isLastPage();
    ck(v,0,0,currVal_0);
  });
}
function View_PaginationControlsComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'ul',[
      [
        'aria-label',
        'Pagination'
      ]
      ,
      [
        'class',
        'ng2-pagination'
      ]
      ,
      [
        'role',
        'navigation'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_5)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_8)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.directionLinks;
    ck(v,3,0,currVal_0);
    const currVal_1:any = import0.ɵnov((<any>v.parent),2).pages;
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.directionLinks;
    ck(v,9,0,currVal_2);
  },(null as any));
}
export function View_PaginationControlsComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'pagination-template',([] as any[]),(null as any),[[
        (null as any),
        'pageChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.PaginationControlsComponent = v.component;
      if (('pageChange' === en)) {
        const pd_0:any = ((<any>co.pageChange.emit($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      import0.ɵdid(368640,[[
        'p',
        4
      ]
    ],0,import3.PaginationControlsDirective,[
      import4.PaginationService,
      import0.ChangeDetectorRef
    ]
    ,{
      id: [
        0,
        'id'
      ]
      ,
      maxSize: [
        1,
        'maxSize'
      ]

    }
    ,{pageChange: 'pageChange'}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PaginationControlsComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import2.PaginationControlsComponent = v.component;
    const currVal_0:any = co.id;
    const currVal_1:any = co.maxSize;
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:boolean = !(co.autoHide && (import0.ɵnov(v,2).pages.length <= 1));
    ck(v,5,0,currVal_2);
  },(null as any));
}
function View_PaginationControlsComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'pagination-controls',([] as any[]),(null as any),(null as any),(null as any),View_PaginationControlsComponent_0,RenderType_PaginationControlsComponent)),
    import0.ɵdid(24576,(null as any),0,import2.PaginationControlsComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const PaginationControlsComponentNgFactory:import0.ComponentFactory<import2.PaginationControlsComponent> = import0.ɵccf('pagination-controls',import2.PaginationControlsComponent,View_PaginationControlsComponent_Host_0,{
  id: 'id',
  maxSize: 'maxSize',
  directionLinks: 'directionLinks',
  autoHide: 'autoHide'
}
,{pageChange: 'pageChange'},([] as any[]));
