// ng2-pagination/dist/pagination-controls.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ng2-pagination/dist/pagination-controls.component';
import * as import3 from 'ng2-pagination/dist/pagination-controls.directive';
import * as import4 from 'ng2-pagination/dist/pagination.service';
var styles_PaginationControlsComponent = ['.ng2-pagination[_ngcontent-%COMP%] {\n  margin-left: 0;\n  margin-bottom: 1rem; }\n  .ng2-pagination[_ngcontent-%COMP%]::before, .ng2-pagination[_ngcontent-%COMP%]::after {\n    content: \' \';\n    display: table; }\n  .ng2-pagination[_ngcontent-%COMP%]::after {\n    clear: both; }\n  .ng2-pagination[_ngcontent-%COMP%]   li[_ngcontent-%COMP%] {\n    -moz-user-select: none;\n    -webkit-user-select: none;\n    -ms-user-select: none;\n    font-size: 0.875rem;\n    margin-right: 0.0625rem;\n    border-radius: 0; }\n  .ng2-pagination[_ngcontent-%COMP%]   li[_ngcontent-%COMP%] {\n    display: inline-block; }\n  .ng2-pagination[_ngcontent-%COMP%]   a[_ngcontent-%COMP%], .ng2-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%] {\n    color: #0a0a0a; \n    display: block;\n    padding: 0.1875rem 0.625rem;\n    border-radius: 0; }\n    .ng2-pagination[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover, .ng2-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover {\n      background: #e6e6e6; }\n  .ng2-pagination[_ngcontent-%COMP%]   .current[_ngcontent-%COMP%] {\n    padding: 0.1875rem 0.625rem;\n    background: #2199e8;\n    color: #fefefe;\n    cursor: default; }\n  .ng2-pagination[_ngcontent-%COMP%]   .disabled[_ngcontent-%COMP%] {\n    padding: 0.1875rem 0.625rem;\n    color: #cacaca;\n    cursor: default; } \n    .ng2-pagination[_ngcontent-%COMP%]   .disabled[_ngcontent-%COMP%]:hover {\n      background: transparent; }\n  .ng2-pagination[_ngcontent-%COMP%]   .ellipsis[_ngcontent-%COMP%]::after {\n    content: \'…\';\n    padding: 0.1875rem 0.625rem;\n    color: #0a0a0a; }\n\n.ng2-pagination[_ngcontent-%COMP%]   .pagination-previous[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]::before, .ng2-pagination[_ngcontent-%COMP%]   .pagination-previous.disabled[_ngcontent-%COMP%]::before { \n  content: \'«\';\n  display: inline-block;\n  margin-right: 0.5rem; }\n\n.ng2-pagination[_ngcontent-%COMP%]   .pagination-next[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]::after, .ng2-pagination[_ngcontent-%COMP%]   .pagination-next.disabled[_ngcontent-%COMP%]::after {\n  content: \'»\';\n  display: inline-block;\n  margin-left: 0.5rem; }\n\n.ng2-pagination[_ngcontent-%COMP%]   .show-for-sr[_ngcontent-%COMP%] {\n  position: absolute !important;\n  width: 1px;\n  height: 1px;\n  overflow: hidden;\n  clip: rect(0, 0, 0, 0); }'];
export var RenderType_PaginationControlsComponent = import0.ɵcrt({
    encapsulation: 0,
    styles: styles_PaginationControlsComponent,
    data: {}
});
function View_PaginationControlsComponent_3(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 4, 'a', [[
                'aria-label',
                'Next page'
            ]
        ], null, [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (import0.ɵnov(v.parent.parent.parent, 2).previous() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        (l()(), import0.ɵted(null, ['\n                Previous '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'show-for-sr'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['page'])),
        (l()(), import0.ɵted(null, ['\n            ']))
    ], null, null);
}
function View_PaginationControlsComponent_4(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 3, 'span', [], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['Previous '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'show-for-sr'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['page']))
    ], null, null);
}
function View_PaginationControlsComponent_2(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 7, 'li', [[
                'class',
                'pagination-previous'
            ]
        ], [[
                2,
                'disabled',
                null
            ]
        ], null, null, null, null)),
        (l()(), import0.ɵted(null, [' \n            '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_3)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n            '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_4)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n        ']))
    ], function (ck, v) {
        var currVal_1 = (1 < import0.ɵnov(v.parent.parent, 2).getCurrent());
        ck(v, 3, 0, currVal_1);
        var currVal_2 = import0.ɵnov(v.parent.parent, 2).isFirstPage();
        ck(v, 6, 0, currVal_2);
    }, function (ck, v) {
        var currVal_0 = import0.ɵnov(v.parent.parent, 2).isFirstPage();
        ck(v, 0, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_6(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 7, 'a', [], null, [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (import0.ɵnov(v.parent.parent.parent, 2).setCurrent(v.parent.context.$implicit.value) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        (l()(), import0.ɵted(null, ['\n                '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'show-for-sr'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['Page'])),
        (l()(), import0.ɵted(null, ['\n                '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [], null, null, null, null, null)),
        (l()(), import0.ɵted(null, [
            '',
            ''
        ])),
        (l()(), import0.ɵted(null, ['\n            ']))
    ], null, function (ck, v) {
        var currVal_0 = v.parent.context.$implicit.label;
        ck(v, 6, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_7(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 7, 'div', [], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['\n                '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'show-for-sr'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['You\'re on page'])),
        (l()(), import0.ɵted(null, ['\n                '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [], null, null, null, null, null)),
        (l()(), import0.ɵted(null, [
            '',
            ''
        ])),
        (l()(), import0.ɵted(null, [' \n            ']))
    ], null, function (ck, v) {
        var currVal_0 = v.parent.context.$implicit.label;
        ck(v, 6, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_5(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 7, 'li', [], [[
                2,
                'current',
                null
            ]
        ], null, null, null, null)),
        (l()(), import0.ɵted(null, ['\n            '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_6)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n            '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_7)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n        ']))
    ], function (ck, v) {
        var currVal_1 = (import0.ɵnov(v.parent.parent, 2).getCurrent() !== v.context.$implicit.value);
        ck(v, 3, 0, currVal_1);
        var currVal_2 = (import0.ɵnov(v.parent.parent, 2).getCurrent() === v.context.$implicit.value);
        ck(v, 6, 0, currVal_2);
    }, function (ck, v) {
        var currVal_0 = (import0.ɵnov(v.parent.parent, 2).getCurrent() === v.context.$implicit.value);
        ck(v, 0, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_9(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 4, 'a', [[
                'aria-label',
                'Next page'
            ]
        ], null, [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (import0.ɵnov(v.parent.parent.parent, 2).next() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        (l()(), import0.ɵted(null, ['\n                Next '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'show-for-sr'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['page'])),
        (l()(), import0.ɵted(null, ['\n            ']))
    ], null, null);
}
function View_PaginationControlsComponent_10(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 3, 'span', [], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['Next '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'show-for-sr'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['page']))
    ], null, null);
}
function View_PaginationControlsComponent_8(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 7, 'li', [[
                'class',
                'pagination-next'
            ]
        ], [[
                2,
                'disabled',
                null
            ]
        ], null, null, null, null)),
        (l()(), import0.ɵted(null, ['\n            '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_9)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n            '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_10)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n        ']))
    ], function (ck, v) {
        var currVal_1 = !import0.ɵnov(v.parent.parent, 2).isLastPage();
        ck(v, 3, 0, currVal_1);
        var currVal_2 = import0.ɵnov(v.parent.parent, 2).isLastPage();
        ck(v, 6, 0, currVal_2);
    }, function (ck, v) {
        var currVal_0 = import0.ɵnov(v.parent.parent, 2).isLastPage();
        ck(v, 0, 0, currVal_0);
    });
}
function View_PaginationControlsComponent_1(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 10, 'ul', [
            [
                'aria-label',
                'Pagination'
            ],
            [
                'class',
                'ng2-pagination'
            ],
            [
                'role',
                'navigation'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['\n\n        '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_2)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n\n        '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_5)),
        import0.ɵdid(401408, null, 0, import1.NgForOf, [
            import0.ViewContainerRef,
            import0.TemplateRef,
            import0.IterableDiffers
        ], { ngForOf: [
                0,
                'ngForOf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n\n        '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_8)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n\n    ']))
    ], function (ck, v) {
        var co = v.component;
        var currVal_0 = co.directionLinks;
        ck(v, 3, 0, currVal_0);
        var currVal_1 = import0.ɵnov(v.parent, 2).pages;
        ck(v, 6, 0, currVal_1);
        var currVal_2 = co.directionLinks;
        ck(v, 9, 0, currVal_2);
    }, null);
}
export function View_PaginationControlsComponent_0(l) {
    return import0.ɵvid(2, [
        (l()(), import0.ɵted(null, ['\n    '])),
        (l()(), import0.ɵeld(0, null, null, 5, 'pagination-template', [], null, [[
                null,
                'pageChange'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('pageChange' === en)) {
                var pd_0 = (co.pageChange.emit($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        import0.ɵdid(368640, [[
                'p',
                4
            ]
        ], 0, import3.PaginationControlsDirective, [
            import4.PaginationService,
            import0.ChangeDetectorRef
        ], {
            id: [
                0,
                'id'
            ],
            maxSize: [
                1,
                'maxSize'
            ]
        }, { pageChange: 'pageChange' }),
        (l()(), import0.ɵted(null, ['\n    '])),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_PaginationControlsComponent_1)),
        import0.ɵdid(8192, null, 0, import1.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵted(null, ['\n    '])),
        (l()(), import0.ɵted(null, ['\n    ']))
    ], function (ck, v) {
        var co = v.component;
        var currVal_0 = co.id;
        var currVal_1 = co.maxSize;
        ck(v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = !(co.autoHide && (import0.ɵnov(v, 2).pages.length <= 1));
        ck(v, 5, 0, currVal_2);
    }, null);
}
function View_PaginationControlsComponent_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'pagination-controls', [], null, null, null, View_PaginationControlsComponent_0, RenderType_PaginationControlsComponent)),
        import0.ɵdid(24576, null, 0, import2.PaginationControlsComponent, [], null, null)
    ], null, null);
}
export var PaginationControlsComponentNgFactory = import0.ɵccf('pagination-controls', import2.PaginationControlsComponent, View_PaginationControlsComponent_Host_0, {
    id: 'id',
    maxSize: 'maxSize',
    directionLinks: 'directionLinks',
    autoHide: 'autoHide'
}, { pageChange: 'pageChange' }, []);
