// ng2-progress-bar/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ng2-progress-bar/index';
import * as import2 from '@angular/common';
class ProgressBarModuleInjector extends import0.ɵNgModuleInjector<import1.ProgressBarModule> {
  _CommonModule_0:import2.CommonModule;
  _ProgressBarModule_1:import1.ProgressBarModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.ProgressBarModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ProgressBarModule_1 = new import1.ProgressBarModule();
    return this._ProgressBarModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.ProgressBarModule)) { return this._ProgressBarModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ProgressBarModuleNgFactory:import0.NgModuleFactory<import1.ProgressBarModule> = new import0.NgModuleFactory<any>(ProgressBarModuleInjector,import1.ProgressBarModule);
const styles_ProgressBar:any[] = ([] as any[]);
export const RenderType_ProgressBar:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ProgressBar,
  data: {}
}
);
export function View_ProgressBar_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n'])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'div',[[
        'class',
        'progress'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',[
      [
        'aria-valuemin',
        '0'
      ]
      ,
      [
        'class',
        'progress-bar'
      ]
      ,
      [
        'role',
        'progressbar'
      ]

    ]
    ,[
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['width']),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'progress-bar-title'
      ]
      ],[[
        8,
        'hidden',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      '% Complete'
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n'])),
    (l()(),import0.ɵted((null as any),['\n']))
  ]
  ,(ck,v) => {
    var co:import1.ProgressBar = v.component;
    const currVal_2:any = ck(v,5,0,(co.getProgress() + '%'));
    ck(v,4,0,currVal_2);
  },(ck,v) => {
    var co:import1.ProgressBar = v.component;
    const currVal_0:any = co.getProgress();
    const currVal_1:any = co.getProgress();
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_3:boolean = !co.title;
    ck(v,7,0,currVal_3);
    const currVal_4:any = co.title;
    ck(v,8,0,currVal_4);
    const currVal_5:any = co.getProgress();
    ck(v,11,0,currVal_5);
  });
}
function View_ProgressBar_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'progress-bar',([] as any[]),(null as any),(null as any),(null as any),View_ProgressBar_0,RenderType_ProgressBar)),
    import0.ɵdid(24576,(null as any),0,import1.ProgressBar,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ProgressBarNgFactory:import0.ComponentFactory<import1.ProgressBar> = import0.ɵccf('progress-bar',import1.ProgressBar,View_ProgressBar_Host_0,{
  value: 'value',
  max: 'max',
  title: 'title'
}
,{},([] as any[]));
