// ng2-progress-bar/index.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as import0 from '@angular/core';
import * as import1 from 'ng2-progress-bar/index';
import * as import2 from '@angular/common';
var ProgressBarModuleInjector = (function (_super) {
    __extends(ProgressBarModuleInjector, _super);
    function ProgressBarModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    Object.defineProperty(ProgressBarModuleInjector.prototype, "_NgLocalization_2", {
        get: function () {
            if ((this.__NgLocalization_2 == null)) {
                (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID)));
            }
            return this.__NgLocalization_2;
        },
        enumerable: true,
        configurable: true
    });
    ProgressBarModuleInjector.prototype.createInternal = function () {
        this._CommonModule_0 = new import2.CommonModule();
        this._ProgressBarModule_1 = new import1.ProgressBarModule();
        return this._ProgressBarModule_1;
    };
    ProgressBarModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.CommonModule)) {
            return this._CommonModule_0;
        }
        if ((token === import1.ProgressBarModule)) {
            return this._ProgressBarModule_1;
        }
        if ((token === import2.NgLocalization)) {
            return this._NgLocalization_2;
        }
        return notFoundResult;
    };
    ProgressBarModuleInjector.prototype.destroyInternal = function () {
    };
    return ProgressBarModuleInjector;
}(import0.ɵNgModuleInjector));
export var ProgressBarModuleNgFactory = new import0.NgModuleFactory(ProgressBarModuleInjector, import1.ProgressBarModule);
var styles_ProgressBar = [];
export var RenderType_ProgressBar = import0.ɵcrt({
    encapsulation: 2,
    styles: styles_ProgressBar,
    data: {}
});
export function View_ProgressBar_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵted(null, ['\n'])),
        (l()(), import0.ɵeld(0, null, null, 12, 'div', [[
                'class',
                'progress'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, ['\n    '])),
        (l()(), import0.ɵeld(0, null, null, 9, 'div', [
            [
                'aria-valuemin',
                '0'
            ],
            [
                'class',
                'progress-bar'
            ],
            [
                'role',
                'progressbar'
            ]
        ], [
            [
                1,
                'aria-valuenow',
                0
            ],
            [
                1,
                'aria-valuemax',
                0
            ]
        ], null, null, null, null)),
        import0.ɵdid(139264, null, 0, import2.NgStyle, [
            import0.KeyValueDiffers,
            import0.ElementRef,
            import0.Renderer
        ], { ngStyle: [
                0,
                'ngStyle'
            ]
        }, null),
        import0.ɵpod(['width']),
        (l()(), import0.ɵted(null, ['\n        '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'progress-bar-title'
            ]
        ], [[
                8,
                'hidden',
                0
            ]
        ], null, null, null, null)),
        (l()(), import0.ɵted(null, [
            '',
            ''
        ])),
        (l()(), import0.ɵted(null, ['\n        '])),
        (l()(), import0.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'sr-only'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵted(null, [
            '',
            '% Complete'
        ])),
        (l()(), import0.ɵted(null, ['\n    '])),
        (l()(), import0.ɵted(null, ['\n'])),
        (l()(), import0.ɵted(null, ['\n']))
    ], function (ck, v) {
        var co = v.component;
        var currVal_2 = ck(v, 5, 0, (co.getProgress() + '%'));
        ck(v, 4, 0, currVal_2);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = co.getProgress();
        var currVal_1 = co.getProgress();
        ck(v, 3, 0, currVal_0, currVal_1);
        var currVal_3 = !co.title;
        ck(v, 7, 0, currVal_3);
        var currVal_4 = co.title;
        ck(v, 8, 0, currVal_4);
        var currVal_5 = co.getProgress();
        ck(v, 11, 0, currVal_5);
    });
}
function View_ProgressBar_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'progress-bar', [], null, null, null, View_ProgressBar_0, RenderType_ProgressBar)),
        import0.ɵdid(24576, null, 0, import1.ProgressBar, [], null, null)
    ], null, null);
}
export var ProgressBarNgFactory = import0.ɵccf('progress-bar', import1.ProgressBar, View_ProgressBar_Host_0, {
    value: 'value',
    max: 'max',
    title: 'title'
}, {}, []);
